### ISSUE TITLE
regression: `S.true & 5` fails expecting bool or Boolean

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#14708]
Good: `S.true & 1` is `S.true` and `S.true & 0` is `S.false`.

Bad?  `S.true & 5` or `S.true & S(5)` used to give `S.true`: after
a9119391e531e1aff8c11a94a0f9c84d36293b4a an error is raised instead.

But after reading over #13427, perhaps we don't care b/c this operation isn't mathematically well-posed.

CC @smichr

**Comment by asmeurer** [comment]
Yes, I remember we decided that trying to interpret integers as booleans was a bad idea. I'm not sure why 0 and 1 are exceptions.

**Comment by asmeurer** [comment]
Ah got it. There's a lot of code that checks `== True`, and this pattern is recommended in the docs IIRC. The alternative is to make sure `x` is sympified and check `x is S.true`, but `x == True` is simpler, and it's easy to accidentally forget to sympify. So I would suggest to leave 0 and 1 be. They shouldn't be considered explicitly supported anywhere, but we also shouldn't necessarily go to lengths  to prevent them from working.
